common-pg/src/main/scala/com/socrata/pg/analyzer2/FuncallSqlizer.scala

```diff
@@ -147,13 +163,45 @@ abstract class FuncallSqlizer[MT <: MetaTypes] extends SqlizerUniverse[MT] {
     }
   }
 
-  protected def sqlizeNormalOrdinaryFuncall(sqlFunctionName: String, prefixArgs: Seq[Doc] = Nil, suffixArgs: Seq[Doc] = Nil) = {
+  def sqlizeNormalOrdinaryFuncall(
+    sqlFunctionName: String,
+    prefixArgs: Seq[Doc] = Nil,
+    suffixArgs: Seq[Doc] = Nil,
+    castType: (Doc, Int) => Option[Doc] = (_, _) => None
+  ) = {
     val funcName = Doc(sqlFunctionName)
     ofs { (e, args, ctx) =>
       assert(args.length >= e.function.minArity)
       assert(e.function.allParameters.startsWith(args.map(_.typ)))
 
-      val sql = (prefixArgs ++ args.map(_.compressed.sql) ++ suffixArgs).funcall(funcName)
+      val castArgs = args.map(_.compressed.sql).zipWithIndex.map { case (arg, idx) =>
+        castType(arg, idx).map { cast =>
+          arg +#+ Doc("::") +#+ cast
```

Comment:
`arg.parenthesized` (it's convention in this code that parens are added at the place where there's an operator)

Reply:
Do you then mean `(arg +#+ Doc("::") +#+ cast).parenthesized`?

Reply:
No, I meant what I said